app/src/main/java/org/hackillinois/android/viewmodel/HomeViewModel.kt

```diff
@@ -11,13 +13,19 @@ class HomeViewModel : ViewModel() {
     lateinit var ongoingEventsLiveData: LiveData<List<Event>>
     lateinit var upcomingEventsLiveData: LiveData<List<Event>>
 
+    private val currentTime: MutableLiveData<Long> = MutableLiveData()
+
     fun init() {
-        ongoingEventsLiveData = eventRepository.fetchEventsHappeningAtTime(System.currentTimeMillis())
-        upcomingEventsLiveData = eventRepository.fetchEventsHappeningInNextHour()
+        ongoingEventsLiveData = Transformations.switchMap(currentTime) {
+            value -> eventRepository.fetchEventsHappeningAtTime(value)
+        }
+        upcomingEventsLiveData = Transformations.switchMap(currentTime) {
+            value -> eventRepository.fetchEventsHappeningInNextHour(value)
+        }
+        currentTime.value = System.currentTimeMillis()
```

Comment:
You could call refresh() here, in case you want to change what refresh does in the future

Reply:
Oh yeah good call